### ISSUE TITLE
prov/sockets memory leak from fi_mr_reg()

### ISSUE THREAD

**Issue by soumagne** [ofiwg/libfabric#5693]
It looks like this is new since 1.9.0. Since the socket code has not really changed it's possibly a side effect of other changes. Here is the trace:
```bash
=================================================================
==28015==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 40 byte(s) in 1 object(s) allocated from:
    #0 0x7f425a285c58 in __interceptor_malloc (/lib64/libasan.so.5+0x10dc58)
    #1 0x7f4259f31d9e in ofi_rbnode_alloc src/tree.c:59
    #2 0x7f4259f323c3 in ofi_rbmap_insert src/tree.c:232
    #3 0x7f4259f5be9c in ofi_mr_map_insert prov/util/src/util_mr_map.c:74
    #4 0x7f4259f6bfaf in sock_regattr prov/sockets/src/sock_mr.c:161
    #5 0x7f4259f6c147 in sock_regv prov/sockets/src/sock_mr.c:200
    #6 0x7f4259f6c1e3 in sock_reg prov/sockets/src/sock_mr.c:211
    #7 0x7f425a0a0660 in fi_mr_reg /opt/libfabric/git/include/rdma/fi_domain.h:328
```
Calling `fi_close()` on the MR handle does not release the memory.

Below is our piece of code that calls into `fi_mr_reg()` and generates that issue:
```C
rc = fi_mr_reg(na_ofi_domain->fi_domain, NULL, UINT64_MAX,
    FI_REMOTE_READ | FI_REMOTE_WRITE | FI_SEND | FI_RECV
    | FI_READ | FI_WRITE, 0 /* offset */, requested_key, 0 /* flags */,
    &na_ofi_domain->fi_mr, NULL /* context */);
```

**Comment by shefty** [comment]
Is this against master?

**Comment by soumagne** [comment]
Yes

**Comment by soumagne** [comment]
Just FYI, I get the same error with rxm/tcp provider

**Comment by shefty** [comment]
Fixed by #5704

**Comment by soumagne** [comment]
Great, confirm it's fixed now! Thank you!
